src/utils/vcs.rs

```diff
@@ -377,6 +388,56 @@ pub fn find_heads(specs: Option<Vec<CommitSpec>>, repos: &[Repo]) -> Result<Vec<
     Ok(rv)
 }
 
+#[test]
+fn test_find_matching_rev() {
+    //TODO: Make sure this works if you pass reference as GitReference::Commit, and pass in a SHA instead of a tag
+    //TODO: Add in more debug just in case this doesn't fix the issue - the customer can send over the debug output.
+    let reference = GitReference::Symbolic("1.9.2");
+    let spec = CommitSpec {
+        repo: String::from("getsentry/sentry-cli"),
+        path: None,
+        rev: String::from("1.9.2"),
+        prev_rev: Some(String::from("1.9.1"))
+    };
+
+    let repos = [
+        Repo {
+            id: String::from("1"),
+            name: String::from("getsentry/sentry-cli"),
+            url: Some(String::from("https://github.com/getsentry/sentry-cli")),
+            provider: RepoProvider { id: String::from("integrations:github"), name: String::from("GitHub") },
+            status: String::from("active"),
+            date_created: chrono::Utc::now()
+        }
+    ];
+
+    let res_with_lightweight_tag = find_matching_rev(reference, &spec, &repos, false);
```

Comment:
I'd suggest having a test for each test-case:
`test_res_with_lightweight_tag`
`test_res_with_annotated_tag`

Reply:
Split them up :)